fastify.js

```diff
@@ -97,12 +97,13 @@ function defaultBuildPrettyMeta (route) {
  */
 function fastify (options) {
   // Options validations
-  options = options || {}
-
-  if (typeof options !== 'object') {
+  if (options && typeof options !== 'object') {
     throw new FST_ERR_OPTIONS_NOT_OBJ()
   }
 
+  // Shallow copy options object to prevent mutations outside of this function
+  options = Object.assign({}, options)
```

Comment:
```suggestion
  } else {
    // Shallow copy options object to prevent mutations outside of this function
    options = Object.assign({}, options)
  }
```